protocols/kad/src/behaviour.rs

```diff
@@ -2551,6 +2551,7 @@ where
             }
             FromSwarm::DialFailure(dial_failure) => self.on_dial_failure(dial_failure),
             FromSwarm::AddressChange(address_change) => self.on_address_change(address_change),
+            FromSwarm::NewExternalAddrOfPeer(_addr) => todo!(),
```

Comment:
This is the place where the new event comes in handy. Here we can add the new address to our routing table.